loopchain/blockchain/blockchain.py

```diff
@@ -227,6 +224,67 @@ def close_blockchain_store(self):
             self._blockchain_store.close()
             self._blockchain_store: KeyValueStore = None
 
+    def check_rollback_possible(self, target_block, start_block=None):
+        """Check if the target block can be reached with the prev_hash of last_block.
+        CAUTION! This method is called recursively.
+
+        :param target_block:
+        :param start_block:
+        :return:
+        """
+        if not start_block:
+            start_block = self.__last_block
```

Comment:
self.__last_block is not thread safe. if blockchain is in competition state in multi thread, call with same args but result is different. i think this is bad pattern. i suggest this function is always called with start_block arg